ui/shared/status/StatusChatImageSizeValidator.qml

```diff
@@ -0,0 +1,72 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtGraphicalEffects 1.0
+import "../../imports"
+import ".."
+
+Item {
+    id: root
+    property bool isValid: true
+    property alias errorMessage: txtValidationError.text
+    property var images: []
+    property var validImages: []
+    readonly property int maxImgSizeBytes: Constants.maxUploadFilesizeMB * 1048576 /* 1 MB in bytes */
+
+    visible: !isValid
+    width: imgExclamation.width + txtValidationError.width + 18
+    height: txtValidationError.height + 14
+
+    onImagesChanged: {
+        let isValid = true
+        root.validImages = images.filter(img => {
+            let size = parseInt(utilsModel.getFileSize(img))
+            const isSmallEnough = size <= maxImgSizeBytes
+            isValid = isValid && isSmallEnough
+            return isSmallEnough
+        })
+        root.isValid = isValid
+    }
+    Rectangle {
+        anchors.fill: parent
+        color: Style.current.background
+        radius: Style.current.halfPadding
+        layer.enabled: true
+        layer.effect: DropShadow {
+            verticalOffset: 3
+            radius: 8
+            samples: 15
+            fast: true
+            cached: true
+            color: "#22000000"
+        }
+
+        SVGImage {
+            id: imgExclamation
+            width: 20
+            height: 20
+            sourceSize.height: height * 2
+            sourceSize.width: width * 2
+            verticalAlignment: Image.AlignVCenter
+            anchors.top: parent.top
+            anchors.left: parent.left
+            anchors.topMargin: 6
+            anchors.leftMargin: 6
+            fillMode: Image.PreserveAspectFit
+            source: "../../app/img/exclamation_outline.svg"
+        }
```

Comment:
Looks like some of this can be made reusable because we seem to do the same `SVGImage`/DropShadow combo in multiple places

Reply:
Good idea. I made the entire validator reusable!